metadata-ingestion/src/datahub/ingestion/source/usage/usage_common.py

```diff
@@ -31,6 +31,65 @@
 TOTAL_BUDGET_FOR_QUERY_LIST = 24000
 
 
+def make_usage_workunit(
+    bucket_start_time: datetime,
+    resource: ResourceType,
+    query_count: int,
+    query_freq: Optional[List[Tuple[str, int]]],
+    user_freq: List[Tuple[str, int]],
+    column_freq: List[Tuple[str, int]],
+    bucket_duration: BucketDuration,
+    urn_builder: Callable[[ResourceType], str],
+    top_n_queries: int,
+    format_sql_queries: bool,
+    total_budget_for_query_list: int = TOTAL_BUDGET_FOR_QUERY_LIST,
+    query_trimmer_string: str = " ...",
+) -> MetadataWorkUnit:
+    top_sql_queries: Optional[List[str]] = None
+    if query_freq is not None:
+        budget_per_query: int = int(total_budget_for_query_list / top_n_queries)
+        top_sql_queries = [
+            trim_query(
+                format_sql_query(query, keyword_case="upper", reindent_aligned=True)
+                if format_sql_queries
+                else query,
+                budget_per_query=budget_per_query,
+                query_trimmer_string=query_trimmer_string,
+            )
+            for query, _ in query_freq
+        ]
+
+    usageStats = DatasetUsageStatisticsClass(
+        timestampMillis=int(bucket_start_time.timestamp() * 1000),
+        eventGranularity=TimeWindowSizeClass(unit=bucket_duration, multiple=1),
+        uniqueUserCount=len(user_freq),
+        totalSqlQueries=query_count,
+        topSqlQueries=top_sql_queries,
+        userCounts=[
+            DatasetUserUsageCountsClass(
+                user=builder.make_user_urn(user_email.split("@")[0]),
+                count=count,
+                userEmail=user_email,
+            )
+            for user_email, count in user_freq
+        ],
+        fieldCounts=[
+            DatasetFieldUsageCountsClass(
+                fieldPath=column,
+                count=count,
+            )
+            for column, count in column_freq
+        ],
+    )
+
+    mcp = MetadataChangeProposalWrapper(
+        entityUrn=urn_builder(resource),
+        aspect=usageStats,
+    )
+
+    return MetadataWorkUnit(id=f"{bucket_start_time.isoformat()}-{resource}", mcp=mcp)
```

Comment:
This will change the workunit id, to putting the timestamp at the front and isoformated, to at the end as an epoch integer. Just want to make sure that's totally fine to do... I don't really know what these workunit ids represent